RSyntaxTextArea/src/main/java/org/fife/ui/rsyntaxtextarea/folding/PythonFoldParser.java

```diff
@@ -80,10 +88,8 @@ else if (leadingWhiteSpaceCount > currentLeadingWhiteSpaceCount) {
 					int endOffs = t.getEndOffset() - 1;
 
 					boolean foundBlock = false;
-					while (!foldStartLeadingWhiteSpaceCounts.isEmpty() &&
+					while (currentFold != null && !foldStartLeadingWhiteSpaceCounts.isEmpty() &&
 							foldStartLeadingWhiteSpaceCounts.peek() >= leadingWhiteSpaceCount) {
-						// IntelliJ can't tell, but it's not possible for currentFold to be
-						// null here
```

Comment:
This shows me to not believe IntelliJ ;)

Reply:
IntelliJ was correct (I think).  When a line continuation was there it blew up (NPE).  The new changes may obviate that, so I could put the comment back and remove the null check if you would like?